docs/proposals/image-promotion.md

```diff
@@ -0,0 +1,152 @@
+
+* Within a single Project
+* Between multiple Projects
+* Between multiple Clusters
+
+The Docker image promotion can be done using the Docker image tags and LABELs.
+You can also use other metadata that OpenShift adds to the Image resource,
+such as annotations or labels, to allow conditional promotion, but this document
+does not describe these advanced scenarios.
+
+Another constraint is that in some cases the image promotion might not be enough,
+and you want to promote a set of images or entire deployment as one single unit
+of promotion.
+
+## Use Cases
+
+### Automatic promotion based on running tests in Jenkins
+
+As a developer of a Rails application, when I update the source code and push it
+to a remote source code repository, I want Jenkins to detect this change and perform a S2I
+build of my application in OpenShift. Depending of the build result, I want Jenkins to
+promote the newly built Docker image by tagging it as "qa-ready" and use OpenShift
+deployments to promote it as "production-ready".
+
+#### Example
+
+As a user I create two DeploymentConfig's:
+
+* "frontend-test"
+* "frontend-production"
+
+The "frontend-test" DeploymentConfig has the ImageChangeTrigger set to watch changes
+in the "qa-ready" ImageStreamTag. This DeploymentConfig also defines the "post" lifecycle
+hook that executes a script in the container deployed and performs additional verification
+and tests.
+
+Example of DeploymentConfig post-lifecycle hook:
+```json
+"strategy": {
+  "type": "Rolling",
+  "rollingParams": {
+    "pre": {
+      "failurePolicy": "Abort",
+      "execNewPod": {
+        "containerName": "test-container",
+        "command": [
+          "/opt/app-root/bin/verify-deployment"
+        ],
+      }
+    }
+  }
+}
+```
+
+Based on the result of the `verify-deployment` command, that runs in the application
+image, the command will promote the image by tagging it as `production-ready` (eg. using
+the `oc tag` command). The command might also notify me about the verification failure.
+
+The "frontend-production" DeploymentConfig has the ImageChangeTrigger set to watch changes
+in the "prod-ready" ImageStreamTag. When a new Docker image with this tag is available,
+it can automatically rolls out a new production deployment. In case you don't want to roll out
+the "frontend-production" automatically, you can avoid specifying the ImageChangeTrigger and
+roll out the new deployment by hand.
+
+### Manual promotion using external Docker registry
+
+As a developer of a Rails application, when I update the source code and OpenShift
+build my image I want to promote the image by passing it throught QA and devops team.
+
+#### Example
+
+* The image is passed to a remote QA team by pushing the image to a shared Docker registry
+* Remote QA team pull the image and drive the testing and verification
+* Remote QA team push the verified image back to the shared Docker registry, tagged as "verified" and with additional sign-off LABEL(s).
```

Comment:
@bparees you're right. It can't be done without adding a new layer to the image. However, I don't see any way for them to add annotation/label on the ImageStreamTag as they don't have access there... 

Would it be possible to have `oc export-image` command that actually exports the Docker image from internal registry (as a tar?) but also include the JSON of the ImageStream(Tag?) ?
And then have `oc import-image` inspect the tar, import the image, but also create the ImageStream(Tag) with all annotations? @smarterclayton